File: scripts/processResultsFile.R
# Takes an results CSV file produced by FastSTI
# and runs a function (default mean) over a
# statistic (description) grouping the result 
# either by the maximum date or all the dates.

processFastSTIResults <- function(filename="results.csv", 
                                  sep=";", 
                                  description="INFECT_RATE_ALIVE",
                                  fun=mean,
                                  header=TRUE,
                                  filter_max_date=TRUE) {
  inp = read.csv(filename, sep=sep, header=header)
  values = inp[inp$description==description,]
  if (filter_max_date) {
    analysisDate = max(as.Date(values$date))
    values = values[as.Date(values$date)==analysisDate,]
  }
  mean_values = aggregate(values$value, by=list(values$name,values$date),FUN=fun)
}

print(processFastSTIResults())

File: scripts/processResultsFile.py
"""Takes a results CSV file produced by FastSTI and runs a function (default
mean) over a statistic (description) grouping the result either by the maximum
date or all the dates.

"""

import pandas

def processFastSTIResults(filename="results.csv",
                          sep=";",
                          description="INFECT_RATE_ALIVE",
                          fun=pandas.DataFrame.mean,
                          filter_max_date=True):
    df = pandas.read_csv(filename, sep=sep)
    values = df[df["description"] == description]
    if filter_max_date:
        analysisDate = df["date"].max()
        values = values[df["date"] == analysisDate]
    return values.groupby("name").apply(fun)

print(processFastSTIResults())
